OpenRA.Mods.Common/Traits/BotModules/SquadManagerBotModule.cs

```diff
@@ -306,34 +306,25 @@ void CreateAttackForce(IBot bot)
 
 		void TryToRushAttack(IBot bot)
 		{
-			var allEnemyBaseBuilder = AIUtils.FindEnemiesByCommonName(Info.ConstructionYardTypes, Player);
-
-			// TODO: This should use common names & ExcludeFromSquads instead of hardcoding TraitInfo checks
-			var ownUnits = activeUnits
-				.Where(unit => unit.IsIdle && unit.Info.HasTraitInfo<AttackBaseInfo>()
-					&& !unit.Info.HasTraitInfo<AircraftInfo>() && !Info.NavalUnitsTypes.Contains(unit.Info.Name) && !unit.Info.HasTraitInfo<HarvesterInfo>()).ToList();
-
-			if (!allEnemyBaseBuilder.Any() || ownUnits.Count < Info.SquadSize)
-				return;
+			// Note that Rush Squad is almost the same as Assuslt Squad at present.
+			// The meaning of this squad now is not rushing everything to a basebuilder with bugs,
+			// instead, it keeps two squads at bay which is isolated to each other for a bot player,
+			// and performance friendly, while Rush Squad tend to aimming at basebuilder.
+			var randomizedSquadSize = Info.SquadSize + World.LocalRandom.Next(Info.SquadSizeRandomBonus);
 
-			foreach (var b in allEnemyBaseBuilder)
+			if (unitsHangingAroundTheBase.Count >= randomizedSquadSize)
```

Comment:
The convention is to return early instead of having long if branches:
```
if (unitsHangingAroundTheBase.Count < randomizedSquadSize))
	return;
```

Reply:
good idea. but I will only apply it to Rush attack now, also Assault Squad also has this problem.